packages/fast-components-react-msft/src/pivot/pivot.tsx

```diff
@@ -0,0 +1,240 @@
+        };
+    }
+
+    private handleTabsUpdate = (activeTabId: string): void => {
+        this.setState({
+            activeId: activeTabId,
+        });
+    };
+
+    private isSelected(element: HTMLElement): boolean {
+        return element.className.includes("active") === true;
+    }
+
+    private updateTabPanelIndex(): void {
+        if (canUseDOM() && this.tabsRef.current && Array.isArray(this.props.items)) {
+            const tabElement: HTMLElement = ReactDOM.findDOMNode(
+                this.tabsRef.current
+            ) as HTMLElement;
+
+            const mytabsArray: HTMLElement[] = Array.prototype.slice.call(
+                tabElement.querySelectorAll("[role='tab']")
+            );
+            this.setState({
+                tabPanelIndex: mytabsArray.findIndex(this.isSelected),
+            });
+        }
+    }
+
+    private generateTabPanelsClassNames(): string {
+        let className: string = get(this.props, "managedClasses.pivot_tabPanels", "");
+        if (this.state.tabPanelIndex === this.prevTabPanelIndex) {
+            className = className;
+        } else if (this.state.tabPanelIndex < this.prevTabPanelIndex) {
+            className = `${className} ${get(
+                this.props,
+                "managedClasses.pivot_tabPanels__previous",
+                "Error"
+            )}`;
+        } else {
+            className = `${className} ${get(
+                this.props,
+                "managedClasses.pivot_tabPanels__next",
+                "Error"
+            )}`;
+        }
+        this.prevTabPanelIndex = this.state.tabPanelIndex;
+        return className;
+    }
+
+    private setActiveIndicatorOffset(): void {
+        if (canUseDOM() && this.tabsRef.current && Array.isArray(this.props.items)) {
+            const tabElement: HTMLElement = ReactDOM.findDOMNode(
+                this.tabsRef.current
+            ) as HTMLElement;
+
+            const mytabs: NodeListOf<Element> = tabElement.querySelectorAll(
+                "[aria-selected='true']"
+            );
+
+            if (mytabs[0] !== undefined && tabElement !== undefined) {
+                const width: number = mytabs[0].getBoundingClientRect().width;
+                const center: number = width / 2;
+                const offsetX: number =
+                    mytabs[0].getBoundingClientRect().left -
+                    tabElement.getBoundingClientRect().left +
+                    center;
+
+                if (offsetX !== this.state.offsetX) {
+                    this.setState({
+                        offsetX,
+                    });
+                }
+            }
+        }
+    }
+
+    /**
+     * Gets the direction of the element
+     */
+    private getLTR(): Direction {
+        if (canUseDOM()) {
+            const tabElement: HTMLElement = ReactDOM.findDOMNode(
+                this.tabsRef.current
+            ) as HTMLElement;
+
+            return !tabElement
+                ? Direction.ltr

```

Comment:
`getComputedStyle` is a pretty heavy function to call every render cycle because it can usually forces style recalculation and sometimes layout reflow (leading to janky animations) - https://gist.github.com/paulirish/5d52fb081b3570c81e3a. 

One solution would be to default this to the *page* direction, which is assigned by adding the `dir` attribute to the `html` elmeent: https://www.w3.org/International/questions/qa-html-dir 

If necessary, would could also add a `dir` prop to the component to provide the opportunity to override the direction on a per-instance basis, but I'm not confident anyone will actually need that capability.

Reply:
chatted offline we will keep this as is for now. Once [this](https://github.com/Microsoft/fast-dna/issues/1438) work gets done we will update